app/src/main/kotlin/io/imhungry/model/RestaurantTagEntity.kt

```diff
@@ -0,0 +1,11 @@
+package io.imhungry.model
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+
+@Entity(tableName = "restaurant_tag",
+    primaryKeys = ["tag", "restaurant_id"])
+data class RestaurantTagEntity(
+    @ColumnInfo(name = "tag") val tag: String,
+    @ColumnInfo(name = "restaurant_id") val id: Int
+)
```

Comment:
You definitely should have an auto-generated ID column, with tag just having a unique constraint. The restaurantId should not be a primary key.